### ISSUE TITLE
http client timeout on Java 12 

### ISSUE THREAD

**Issue by kanglais** [http-kit/http-kit#411]
Java verison
```
openjdk version "12" 2019-03-19
OpenJDK Runtime Environment (build 12+33)
OpenJDK 64-Bit Server VM (build 12+33, mixed mode, sharing)
```
Lein version
`Leiningen 2.9.1 on Java 12 OpenJDK 64-Bit Server VM
`
http-kit versions 2.3.0, 2.4.0-alpha3

I expect the request to succeed but return 411 status, instead it blows up with a Timeout Exception.

I was trying to find a fix, any pointers on best places to start?

Here is a minimal example. 

```
(ns app.core
  (:require [org.httpkit.client :as http])
  )

(defn options [body] {:timeout 2       
                      :body body})

(defn create-resource [request-body]
  (http/post "https://google.com" (options request-body)))

(defn -main
  [& args]
  (let [test-resp (create-resource "{\"foo\": 1}")]
    (println @test-resp)))
```

```
/Library/Java/JavaVirtualMachines/openjdk-12.jdk/Contents/Home/bin/java -Dfile.encoding=UTF-8 -XX:-OmitStackTraceInFastThrow -Dclojure.compile.path=/Users/kellieenglish/wonderkind/app/target/classes -Dapp.version=0.1.0-SNAPSHOT -Dclojure.debug=false "-javaagent:/Users/kellieenglish/Applications/IntelliJ IDEA CE.app/Contents/lib/idea_rt.jar=64368:/Users/kellieenglish/Applications/IntelliJ IDEA CE.app/Contents/bin" -classpath /Users/kellieenglish/wonderkind/app/test:/Users/kellieenglish/wonderkind/app/src:/Users/kellieenglish/wonderkind/app/dev-resources:/Users/kellieenglish/wonderkind/app/resources:/Users/kellieenglish/wonderkind/app/target/classes:/Users/kellieenglish/.m2/repository/org/clojure/clojure/1.10.0/clojure-1.10.0.jar:/Users/kellieenglish/.m2/repository/org/clojure/spec.alpha/0.2.176/spec.alpha-0.2.176.jar:/Users/kellieenglish/.m2/repository/org/clojure/core.specs.alpha/0.2.44/core.specs.alpha-0.2.44.jar:/Users/kellieenglish/.m2/repository/http-kit/http-kit/2.3.0/http-kit-2.3.0.jar:/Users/kellieenglish/.m2/repository/nrepl/nrepl/0.6.0/nrepl-0.6.0.jar:/Users/kellieenglish/.m2/repository/clojure-complete/clojure-complete/0.2.5/clojure-complete-0.2.5.jar clojure.main -i /private/var/folders/dd/dn0rwdy932z0tr6lp5vzvh_00000gn/T/form-init850616921077774897.clj
Thu Aug 29 13:05:46 CEST 2019 [client-loop] ERROR - select exception, should not happen
java.lang.IllegalStateException: Client/Server mode has not yet been set.
	at java.base/sun.security.ssl.SSLEngineImpl.beginHandshake(SSLEngineImpl.java:98)
	at org.httpkit.client.HttpClient.finishConnect(HttpClient.java:395)
	at org.httpkit.client.HttpClient.run(HttpClient.java:472)
	at java.base/java.lang.Thread.run(Thread.java:835)
Thu Aug 29 13:05:46 CEST 2019 [client-loop] ERROR - select exception, should not happen
java.lang.IllegalStateException: Client/Server mode has not yet been set.
	at java.base/sun.security.ssl.SSLEngineImpl.beginHandshake(SSLEngineImpl.java:98)
	at org.httpkit.client.HttpClient.finishConnect(HttpClient.java:395)
	at org.httpkit.client.HttpClient.run(HttpClient.java:472)
	at java.base/java.lang.Thread.run(Thread.java:835)
{:opts {:timeout 2, :body {"foo": 1}, :method :post, :url https://google.com}, :error #error {
 :cause idle timeout: 2ms
 :via
 [{:type org.httpkit.client.TimeoutException
   :message idle timeout: 2ms
   :at [org.httpkit.client.HttpClient clearTimeout HttpClient.java 120]}]
 :trace
 [[org.httpkit.client.HttpClient clearTimeout HttpClient.java 120]
  [org.httpkit.client.HttpClient run HttpClient.java 481]
  [java.lang.Thread run Thread.java 835]]}}

Process finished with exit code 0
```

**Comment by vincentjames501** [comment]
2.4.0-alpha3 should resolve the `java.lang.IllegalStateException: Client/Server mode has not yet been set.` on newer Java versions.

In your output you are using 2.3.0:

`/Users/kellieenglish/.m2/repository/http-kit/http-kit/2.3.0/http-kit-2.3.0.jar`

**Comment by RokLenarcic** [comment]
2.4.0-alpha3 does not resolve this

**Comment by zerg000000** [comment]
or you can just use `org.httpkit.sni-client` in `2.4.0-alpha6`

**Comment by ptaoussanis** [comment]
Closing as part of issue triage since I believe this is resolved for <= `2.7.0-beta3`.
If this is still relevant, please feel free to re-open.
